cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -549,7 +550,17 @@ table_with_metadata reader::impl::read_chunk_internal(read_mode mode)
   out_columns.reserve(_output_buffers.size());
 
   // no work to do (this can happen on the first pass if we have no rows to read)
-  if (!has_more_work()) { return finalize_output(out_metadata, out_columns); }
+  if (!has_more_work()) {
+    // Compute number of rows per source if no AST filters
+    if (not _expr_conv.get_converted_expr().has_value()) {
```

Comment:
Empty DF case. Simply initialize the `out_metadata.num_rows_per_source` to all zeros.